lib/parameters.js

```diff
@@ -14,7 +14,13 @@ function generateVarConversion(ctx, overload, i, parent, errPrefix, targetIdx =
   const requires = new utils.RequiresMap(ctx);
   const idlType = overload.typeList[i];
   // Always (try to) force-convert dictionaries
-  const optional = overload.optionalityList[i] === "optional" && !ctx.dictionaries.has(idlType.idlType);
+  const isDefaultedDictionary = overload.operation.arguments[i].default &&
+                                overload.operation.arguments[i].default.type === "dictionary";
+  if (isDefaultedDictionary && !ctx.dictionaries.has(idlType.idlType)) {
+    throw new Error(
+      `The dictionary ${idlType.idlType} was referenced in an argument list, but doesn't exist`);
```

Comment:
Couldn't it also be another class of errors: the default value is incorrectly set to be `{}`?

Reply:
Fair point. So, maybe something like "The argument X was defaulted to {}, but there is no dictionary named Y".